### PR TITLE
Fixes #24735: Add an option to "update only" user in the Users technique

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/24735

### CODE REVIEW

**Comment by ncharles** [comment]
this is a wip, do not test; I don't know why it created a PR

**Comment by ncharles** [comment]
PR updated with a new commit

**Comment by ncharles** [comment]
PR rebased

**Comment by ncharles** [comment]
PR rebased

**Comment by ncharles** [comment]
PR updated with a new commit

**Comment by ncharles** [comment]
PR updated with a new commit

File: techniques/systemSettings/userManagement/userManagement/10/userManagement.st [hunk 1]
```diff
@@ -127,32 +127,29 @@ bundle agent check_usergroup_user_parameters_&RudderUniqueID&
...
-      "usermanagement_user_remove_${usergroup_user_index}" expression => strcmp("${usergroup_user_action[${usergroup_user_index}]}","remove");
-
-      "usermanagement_user_checkpres_${usergroup_user_index}" expression => strcmp("${usergroup_user_action[${usergroup_user_index}]}","checkhere");
-
-      "usermanagement_user_checkabs_${usergroup_user_index}" expression => strcmp("${usergroup_user_action[${usergroup_user_index}]}","checknothere");
+      "usermanagement_user_add_${usergroup_user_index}"        expression => strcmp("${usergroup_user_action[${usergroup_user_index}]}","add");
+      "usermanagement_user_updateonly_${usergroup_user_index}" expression => strcmp("${usergroup_user_action[${usergroup_user_index}]}","updateonly");
+      "usermanagement_user_remove_${usergroup_user_index}"     expression => strcmp("${usergroup_user_action[${usergroup_user_index}]}","remove");
+      
+      # If we ask to update only and user don't exist, then we won't do anything
```
Comment (Fdall) [inline techniques/systemSettings/userManagement/userManagement/10/userManagement.st:139]: the user does not exist

File: techniques/systemSettings/userManagement/userManagement/10/userManagement.st [hunk 2]
```diff
@@ -127,32 +127,29 @@ bundle agent check_usergroup_user_parameters_&RudderUniqueID&
...
+                                                                              "usermanagement_user_exists_${usergroup_user_index}",
+                                                                               or(
+                                                                                 "usermanagement_user_add_${usergroup_user_index}",
+                                                                                 "usermanagement_user_updateonly_${usergroup_user_index}"
+                                                                               )
+                                                                             );
 
+    # Actions
+    any::
+      # NOTE: usergroup_user_home_perso is true when we use the default home, and thus usermanagement_user_pershome_ is true when we don't want tu use the default home
```
Comment (Fdall) [inline techniques/systemSettings/userManagement/userManagement/10/userManagement.st:152]: to use

File: techniques/systemSettings/userManagement/userManagement/10/userManagement.ps1.st [hunk 3]
```diff
@@ -91,23 +96,36 @@ function check_usergroup_user_parameters_&RudderUniqueID& {
...
+        $null = Rudder-Report-NA  @passwordParams -message "User does not exists"
       }
 
       #Unix section reporting
-      $unixParams = $commonParams + @{
-          ComponentName = "Home directory"
+      $primaryGroup = $commonParams + @{
+          ComponentName = "User primary group"
+          ComponentKey = $logins[$i]
+      }
```
Comment (Fdall) [inline techniques/systemSettings/userManagement/userManagement/10/userManagement.ps1.st:108]: You can create a `naParams` variable based on the `commonParams` variable and extend it with the message and componentKey. Something like:
``
$naParams = $commonParams  + @{
  ComponentKey = $logins[$i]
  Message = 'Not applicable on Windows'
}
$null = Rudder-Report-NA @naParams -ComponentName 'User primary group'
$null = Rudder-Report-NA @naParams -ComponentName 'User full name'
$null = Rudder-Report-NA @naParams -ComponentName 'User default shell'
$null = Rudder-Report-NA @naParams -ComponentName 'Home Directory'

**Comment by ncharles** [comment]
PR updated with a new commit

**Review by Fdall — APPROVED** [review]

**Pull request merged**